File: C++/题目1486：False coin.cpp

#include <stdio.h>
#include <string.h>
const int maxn = 1002;
int arrayA[maxn];
int arrayB[maxn];
char symbol[2];
int n,i,j,k;
int abs(int x){
	return x < 0 ? -x : x;
}
void initArray(){
	for(i = 1; i < n+1; i++){
		arrayA[i] = 0;
		arrayB[i] = 1;
	}
}
int main(){
    while(scanf("%d%d",&n,&k) != EOF){
		int num = 0;
		initArray();
		for(i = 0; i < k ; i++){
			int first;
			scanf("%d",&first);
			int *tempArrayA = new int[first];
            int *tempArrayB = new int[first];
			for (j = 0; j < first; j++) {        
				scanf("%d",&tempArrayA[j]);
			}
			for (j = 0; j < first; j++) {        
				scanf("%d",&tempArrayB[j]);
			}
			scanf("%s",symbol);
                
			if(symbol[0] == '='){
				for(j = 0; j < first; j++){
					arrayB[tempArrayA[j]] = 0;
					arrayB[tempArrayB[j]] = 0;
				}
			}else if(symbol[0] == '>'){
				num++;
				for(j = 0; j < first; j++){
					arrayA[tempArrayA[j]]++;
					arrayA[tempArrayB[j]]--;
				}
			}else if(symbol[0] == '<'){
				num++;
				for(j = 0; j < first; j++){
					arrayA[tempArrayA[j]]--;
					arrayA[tempArrayB[j]]++;
				}
			}
		}
		int maxId = 0;
		int count = 0;
		for(j = 1; j < n+1; j++){
			if(arrayB[j] == 1 && abs(arrayA[j]) == num){
				maxId = j;
				count++;
			}
		}
		printf("%d\n",count == 1 ? maxId : 0);
    }
    return 0;
}
/**************************************************************
	Problem: 1486
	User: wangzhenqing
	Language: C++
	Result: Accepted
	Time:30 ms
	Memory:1428 kb
****************************************************************/


                        

File: Java/题目1486：False coin.java

// 题目1486：False coin

/**
 * @author:wangzq
 * @email:wangzhenqing1008@163.com
 * @date:2015-06-30 11:01:55
 * @url:http://ac.jobdu.com/problem.php?pid=1486
 * 解题思路参考csdn:http://blog.csdn.net/u013027996/article/details/17339893
 */

import java.util.Arrays;
import java.util.Scanner;

public class Main {
    /*
     * 1486
     */
    public static void main(String[] args) throws Exception {
        Scanner scanner = new Scanner(System.in);
        while (scanner.hasNext()) {
            int n = scanner.nextInt();
            int k = scanner.nextInt();
            int num = 0;
            int arrayA[] = new int[n + 1];
            int arrayB[] = new int[n + 1];
            Arrays.fill(arrayB, 1);
            while (k > 0) {
                int first = scanner.nextInt();
                int tempArrayA[] = new int[first];
                int tempArrayB[] = new int[first];
                for (int i = 0; i < first; i++) {
                    tempArrayA[i] = scanner.nextInt();
                }
                for (int i = 0; i < first; i++) {
                    tempArrayB[i] = scanner.nextInt();
                }
                String symbol = scanner.next();
                if (symbol.equals("=")) {
                    for (int i = 0; i < first; i++) {
                        arrayB[tempArrayA[i]] = 0;
                        arrayB[tempArrayB[i]] = 0;
                    }
                } else if (symbol.equals(">")) {
                    num++;
                    for (int i = 0; i < first; i++) {
                        arrayA[tempArrayA[i]]++;
                        arrayA[tempArrayB[i]]--;
                    }
                } else if (symbol.equals("<")) {
                    num++;
                    for (int i = 0; i < first; i++) {
                        arrayA[tempArrayA[i]]--;
                        arrayA[tempArrayB[i]]++;
                    }
                }
                k--;
            }
            int max = 0;
            int maxId = 0;
            int j = 1;
            while (j < n + 1) {
                int tempNum = Math.abs(arrayA[j]);
                if (arrayB[j] == 1 && tempNum == num) {
                    max++;
                    maxId = j;
                }
                j++;
            }
            System.out.println(max != 1 ? 0 : maxId);
        }
    }
}

/**************************************************************
 Problem: 1486
 User: wzqwsrf
 Language: Java
 Result: Accepted
 Time:640 ms
 Memory:106776 kb
 ****************************************************************/




